src/Stratis.Bitcoin.Features.Miner.Tests/PosMintingTest.cs

```diff
@@ -351,6 +352,100 @@ public void GetNetworkWeight_UsesLast73Blocks_CalculatedNetworkWeightUsingLatest
 
             Assert.Equal(4607763.9659653762, weight);
         }
+        
+        [Fact]
+        public void CoinstakeAge_BeforeActivation_Testnet()
+        {
+            bool suitable = this.WasUtxoSelectedForStaking(Network.StratisTest, 1000, 1000 - 8); // utxo depth is 9, mining block at 10
+            Assert.True(suitable);
+
+            bool suitable2 = this.WasUtxoSelectedForStaking(Network.StratisTest, 1000, 1000 - 7); // utxo depth is 8, mining block at 9
+            Assert.False(suitable2);
+        }
+
+        [Fact]
+        public void CoinstakeAge_AfterActivation_Testnet()
+        {
+            bool suitable = this.WasUtxoSelectedForStaking(Network.StratisTest, 450000, 450000 - 18);
+            Assert.True(suitable);
+
+            bool suitable2 = this.WasUtxoSelectedForStaking(Network.StratisTest, 450000, 450000 - 17);
+            Assert.False(suitable2);
+        }
+
+        [Fact]
+        public void CoinstakeAge_AtTheActivation_Testnet()
+        {
+            bool suitable1 = this.WasUtxoSelectedForStaking(Network.StratisTest, 436000 - 2, 436000 - 10); // mining block before activation
+            Assert.True(suitable1);
+
+            bool suitable2 = this.WasUtxoSelectedForStaking(Network.StratisTest, 436000 - 1, 436000 - 19); // mining activation block
+            Assert.True(suitable2);
+
+            bool suitable3 = this.WasUtxoSelectedForStaking(Network.StratisTest, 436000 - 1, 436000 - 18); // mining activation block
+            Assert.False(suitable3);
+        }
+        
+        [Fact]
+        public void CoinstakeAge_BeforeActivation_Mainnet()
+        {
+            bool suitable = this.WasUtxoSelectedForStaking(Network.StratisMain, 1000, 1000 - 48); // utxo depth is 49, mining block at 50
+            Assert.True(suitable);
+
+            bool suitable2 = this.WasUtxoSelectedForStaking(Network.StratisMain, 1000, 1000 - 47); // utxo depth is 48, mining block at 49
+            Assert.False(suitable2);
+        }
+
+        [Fact]
+        public void CoinstakeAge_AfterActivation_Mainnet()
+        {
+            bool suitable = this.WasUtxoSelectedForStaking(Network.StratisMain, 950000, 950000 - 498);
+            Assert.True(suitable);
+
+            bool suitable2 = this.WasUtxoSelectedForStaking(Network.StratisMain, 950000, 950000 - 497);
+            Assert.False(suitable2);
+        }
+
+        [Fact]
+        public void CoinstakeAge_AtTheActivation_Mainnet()
+        {
+            bool suitable1 = this.WasUtxoSelectedForStaking(Network.StratisMain, 940000 - 2, 940000 - 50); // mining block before activation
```

Comment:
magic numbers and comments make this difficult to understand what it is actually checking. Can you give this more meaning by giving all the magic numbers some meaning with variable names or constants.

Reply:
Fair enough. Added comments.